dds/idl/be_init.cpp

```diff
@@ -56,7 +58,10 @@ BE_post_init(char*[], long)
   ACE_CString included;
   DRV_add_include_path(included, include_dds.c_str(), 0, true);
 
-  if (idl_global->idl_version_ >= IDL_VERSION_4) {
+  if (idl_global->idl_version_ < IDL_VERSION_4) {
+    idl_global->ignore_files_ = true; // Exit without parsing files
+    be_global->error("opendds_idl requires IDL version to be 4 or greater");
```

Comment:
When would this error occur?  Is the user explicitly setting a version?  Should that be allowed, or should the BE just set the version it wants to use?

Reply:
> When would this error occur? Is the user explicitly setting a version?

Yes.

> Should that be allowed, or should the BE just set the version it wants to use?

That is a good question. My goal in the FE was that the BE and the user would to be able to have some sort of control over IDL version, which is why I have a "default default" version for TAO_IDL_FE to set, a default version that the BE can set, and the version that the user can set. However it doesn't currently cover a minimum version and I came to conclusion that `opendds_idl` should probably ultimately restrict itself to >=IDL4 given we would like people to use annotations. However this isn't strictly necessary to enforce that at the moment, so I'm willing to change it back.

Reply:
I guess what I was missing when I wrote my comment was that the FE is already parsing `--idl-version` and we can't turn it off from the BE.  So that we don't expand the scope of this change to the FE, I'm OK leaving that for now.  We could change it later -- it would be good to have more modular functionality that the BE (really the driver) can decide to invoke from the FE.  There are some other tao_idl issues here that are out of scope for this PR.

Since parsing version 4 is backwards-compatible, I can't think of a reason it wouldn't be set that way (aside from accidental misconfiguration).